va/validation-authority.go

```diff
@@ -212,12 +208,10 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 		scheme = "https"
 		port = va.httpsPort
 	}
-	portString := strconv.Itoa(port)
-	hostPort := net.JoinHostPort(host, portString)
 
 	url := &url.URL{
 		Scheme: scheme,
-		Host:   hostPort,
+		Host:   host,
```

Comment:
Yes, it is important to send a Host header, but the Go http library already does that for us. The Host field there is just for overriding the default (["use URL.Host"](https://golang.org/pkg/net/http/#Request)) which is already what we want.

This particular URL always leaves off the port number because we always start validation using the default port: we have a "port" variable, but it is always the default port for the current scheme.

The only time we fetch from a non-default port is when following a redirect to a URL with `":"` in its host, and that case is already handled correctly by the Go http lib because we don't munge `req.URL.Host` or `req.Host` during redirects.